larky/src/main/resources/stdlib/xml/etree/ElementTree.star

```diff
@@ -941,72 +955,68 @@ def _namespaces(elem, default_namespace=None):
                 add_qname(tag)
         elif tag != None and tag != Comment and tag != PI:
             _raise_serialization_error(tag)
-        for key, value in elem.items():
+        for key, value in current.items():
             if types.is_instance(key, QName):
                 key = key.text
             if key not in qnames:
                 add_qname(key)
             if types.is_instance(value, QName) and value.text not in qnames:
                 add_qname(value.text)
-        text = elem.text
+        text = current.text
         if types.is_instance(text, QName) and text.text not in qnames:
             add_qname(text.text)
+        qu.extend(current._children)
     return qnames, namespaces
 
+def prepare_elems_for_serialize(tops_level_elems):
+    new_elems = []
+    for el in tops_level_elems:
+        new_elems.append(el)
+        qu = el._children[0:]
+        for _ in range(_WHILE_LOOP_EMULATION_ITERATION):
+            if not qu:
+                break
+            else:
+                current = qu.pop()
+                new_elems.append(current)
+                qu.extend(current._children)
+    return new_elems
 
 def _serialize_xml(
-    write, elem, qnames, namespaces, short_empty_elements, **kwargs
+    write, root_elem, qnames, namespaces, short_empty_elements, **kwargs
```

Comment:
when we say "root_elem" do we really mean we can only serialize an entire xml tree, not an individual node?

Reply:
no, I have showed it in the latest update that it works on child element. I changed it to start_elem, it was for me to differentiate variables as I am also using 'elem' in function inner statement